src/Socket.cpp

```diff
@@ -240,6 +255,31 @@ MessagePtr Socket::takeNextMessage()
     }
 }
 
+MessagePtr Socket::takeNextMessageBlocking()
+{
+
+    // Keep requirest wait until a new message received
+    std::unique_lock<std::mutex> lk(d->receiveQueueMutexBlock);
+
+    while (d->message_ready == false)
+    {
+        d->socket_block_condition_variable.wait(lk);
+    }
+
+    std::lock_guard<std::mutex> lock(d->receiveQueueMutex);
+    if(d->receiveQueue.size() > 0)
+    {
+        MessagePtr next = d->receiveQueue.front();
+        d->receiveQueue.pop_front();
+        return next;
+    }
+    else
+    {
+        d->message_ready = false;
+        return nullptr;
+    }
```

Comment:
Just call `takeNextMessage()` instead of all this. You can set `message_ready` to `false` afterwards by checking for `nullptr`, if that's necessary.

Reply:
I think I did not understand well the expected change.

Reply:
This is basically what Lipu did in the subsequent commits, so you can check the diff to see what they meant :-)